src/time.rs

```diff
@@ -15,36 +14,73 @@ const FORMAT_DATETIME_LONG: &str = "%Y-%m-%dT%H:%M:%S%.fZ";
 
 pub const FORMAT_DATE_AND_TIME: &str = "%Y-%m-%d %H:%M";
 
+pub trait TimeProvider: Send + Sync {
+    fn now(&self, tz: Tz) -> DateTime<Tz>;
+    fn today(&self, tz: Tz) -> NaiveDate {
+        self.now(tz).date_naive()
+    }
+}
```

Comment:
I really like the use of this trait 👍  it's just the Arc that I would like to avoid.